ghcide/src/Development/IDE/Core/PositionMapping.hs

```diff
@@ -123,6 +124,10 @@ mkDelta cs = foldl' applyChange idDelta cs
 addDelta :: PositionDelta -> PositionMapping -> PositionMapping
 addDelta delta (PositionMapping pm) = PositionMapping (composeDelta delta pm)
 
+-- | Add a old delta onto a Mapping k n to make a Mapping (k - 1) n
```

Comment:
Could you expand on the documentation for this as well as `addDelta` clarifying which is appropriate to use in which situation with something like the example you gave in the PR?

Reply:
done, the added haddock comment should be clarifying things out.